doc/texture_modifiers.adoc

```diff
@@ -0,0 +1,433 @@
+* `frame`: Current animation frame ("crack progression").
+
+NOTE: This always scales the crack to the size of the base texture (or the tiles of the base texture, if `tilesize` is provided).
+
+===== `<base>^[sheet:<w>x<h>:<x>,<y>`
+
+* `w`, `h`: Tile dimensions (positive integers)
+* `x`, `y`: Tile position, 0-indexed
+
+Retrieves the tile at position `x`, `y`. Can be used to retrieve single pixels by setting `w`, `h` to `1`.
+
+WARNING: Setting `w` or `h` to `0` will trigger a floating point exception, crashing the client.
+
+===== `<base>^[multiply:<color>`
+
+* `color` is a `ColorString`
+
+Multiplies the RGB values of the base texture per pixel with the RGB values of `color`; the alpha value of `color` is ignored.
+
+===== `<base>^[colorize:<color>[:<ratio>]`
+
+* `color` is a `ColorString`
+* `ratio` is an optional integer from `0` to `255` or the string `alpha`
+
+Interpolates between `color` and the pixel colors of the base texture as specified by the `ratio`:
+
+* Defaults to the alpha of `color` if omitted;
+* If it is an integer from `0` (only pixel color) to `255` (only `color`), it is directly used as interpolation ratio:
+  The resulting color of a pixel is `ratio` times `color` plus `(255 - ratio)` times pixel color;
+* If it is the string `alpha`, the texture pixel's alpha value determines the `ratio` per pixel
+
+===== `<base>^[mask:<texture>`
+
+* `texture` is an escaped texture modifier
+
+Applies _bitwise and_ (`bit.band`) to all RGBA values of `texture` and the base texture.
+
+The dimensions of the resulting texture are determined by the base texture.
+
+If a pixel of the base texture is out of bounds on `texture`, it is preserved.
+
+Masking is associative and commutative if all involved textures have the same dimensions.
+
+===== `<base>^[lowpart:<percent>:<texture>`
+
+* `percent` is an integer from `0` to `100`
+* `texture` is an escaped texture modifier
+
+Overlays the lower `percent` part of `texture` on the base texture `x`.
+
+TIP: Use `blank.png` as base texture `x` if you do not want a background.
+
+==== Base Texture Generators
+
+These modifiers do not accept a base texture as they generate one from their arguments.
+
+===== `[png:<data>`
+
+* `data` is a base64-encoded PNG bytestring
+
+Creates a texture from an embedded base64-encoded PNG image.
+
+`data` can be built by combining `minetest.encode_base64` and `minetest.encode_png`:
+
+[source,lua]
+====
+local function embedded_png(width, height, data)
+	return "[png:" .. minetest.encode_base64(minetest.encode_png(width, height, data))
+end
+====
+
+WARNING: Do not use this for large textures. If used as an object texture,
+the texture modifier will get sent arbitrarily often, putting a strain on the network.
+
+TIP: Consider using other texture modifiers cleverly or using dynamic media instead.
+
+===== `[combine:<w>x<h>:<textures>`
```

Comment:
since we target latest, probably can ignore this, but https://github.com/minetest/minetest/issues/12209 where krock/smalljoke broke using it by itself for older versions may be worth mentioning?